packages/vscode-extension/src/project/bridge.ts

```diff
@@ -0,0 +1,104 @@
+import { Disposable } from "vscode";
+import { NavigationRoute } from "../common/Project";
+
+export interface RadonInspectorBridgeEvents {
+  appReady: [];
+  navigationChanged: [{ displayName: string; id: string }];
+  navigationRouteListUpdated: [NavigationRoute[]];
+  fastRefreshStarted: [];
+  fastRefreshComplete: [];
+  openPreviewResult: [{ previewId: string; error?: string }];
+  inspectData: [{ id: number }];
+  devtoolPluginsChanged: [{ plugins: string[] }];
+  pluginMessage: [{ pluginId: string; type: string; data: any }];
+  isProfilingReact: [boolean];
+}
+
+export interface RadonInspectorBridge {
```

Comment:
Like mentioned before, I'd rename this to something like `RadonApplicationBridge` to avoid confusing this as something Inspector-specific.